File: models/AbstractItem.java
package models;

import java.io.Serializable;

public class AbstractItem extends Model implements Serializable
{

    protected int itemCode;
    protected int cost;

    public AbstractItem(int index)
    {
	super(index);
    }

    public int getCost()
    {
	return cost;
    }

    public void setCost(int cost)
    {
	this.cost = cost;
    }

    public int getItemCode()
    {
	return itemCode;
    }

    public void setItemCode(int itemCode)
    {
	this.itemCode = itemCode;
    }

    public AbstractItem(AbstractItem i, int listIndex)
    {
	this.name = i.name;
	this.gameIndex = i.gameIndex;
	this.itemCode = i.itemCode;
	this.cost = i.cost;
	this.namePointer = i.namePointer;
    }

}

File: src/application/models/AbstractItem.kt
package application.models

import java.io.Serializable

open class AbstractItem : Model, Serializable
{

	var itemCode: Int = 0
	var cost: Int = 0


	constructor(index: Int) : super(index)
	constructor(i: AbstractItem) : super(i)
	{
		this.itemCode = i.itemCode
		this.cost = i.cost
	}

	fun getCostText(): MutableList<Int>
	{
		var list = mutableListOf<Int>()
		var temp = cost
		while (temp > 0)
		{
			list.add(temp % 10)
			temp /= 10
		}
		list.reverse()

		return list
	}

}